pkg/controller/perconaservermongodbrestore/physical.go

```diff
@@ -949,3 +1013,73 @@ func (r *ReconcilePerconaServerMongoDBRestore) waitForPBMOperationsToFinish(ctx
 
 	return nil
 }
+
+func (r *ReconcilePerconaServerMongoDBRestore) checkIfPBMAgentsReadyForPhysicalRestore(ctx context.Context, cluster *psmdbv1.PerconaServerMongoDB) (bool, error) {
+	log := logf.FromContext(ctx)
+
+	stdoutBuf := &bytes.Buffer{}
+	stderrBuf := &bytes.Buffer{}
+
+	err := retry.OnError(anotherOpBackoff, func(err error) bool {
+		return strings.Contains(err.Error(), "unable to upgrade connection")
+	}, func() error {
+		stdoutBuf.Reset()
+		stderrBuf.Reset()
+
+		pod := corev1.Pod{}
+		nn := types.NamespacedName{Name: cluster.Spec.Replsets[0].PodName(cluster, 0), Namespace: cluster.Namespace}
+		if err := r.client.Get(ctx, nn, &pod); err != nil {
+			return errors.Wrap(err, "get pod")
+		}
+
+		container := "mongod"
+		pbmBinary := "/opt/percona/pbm"
+		for _, c := range pod.Spec.Containers {
+			if c.Name == naming.ContainerBackupAgent {
+				container = c.Name
+				pbmBinary = "pbm"
```

Comment:
I think we can break the loop here if we find the backup agent container, no need to continue

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/1861/commits/b5ff21ce2bedc65d6ce7807638eb52f7ce1ef00b